File: openbis_api/source/java/ch/ethz/sis/openbis/generic/asapi/v3/dto/semanticannotation/search/SemanticAnnotationSearchCriteria.java
/*
 * Copyright 2014 ETH Zuerich, CISD
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package ch.ethz.sis.openbis.generic.asapi.v3.dto.semanticannotation.search;

import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.search.AbstractObjectSearchCriteria;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.search.PermIdSearchCriteria;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.search.SearchCriteriaToStringBuilder;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.search.SearchOperator;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.entitytype.search.EntityTypeSearchCriteria;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.property.search.PropertyAssignmentSearchCriteria;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.property.search.PropertyTypeSearchCriteria;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.semanticannotation.id.ISemanticAnnotationId;
import ch.systemsx.cisd.base.annotation.JsonObject;

/**
 * @author pkupczyk
 */
@JsonObject("as.dto.semanticannotation.search.SemanticAnnotationSearchCriteria")
public class SemanticAnnotationSearchCriteria extends AbstractObjectSearchCriteria<ISemanticAnnotationId>
{

    private static final long serialVersionUID = 1L;

    public SemanticAnnotationSearchCriteria()
    {
    }

    public PermIdSearchCriteria withPermId()
    {
        return with(new PermIdSearchCriteria());
    }

    public EntityTypeSearchCriteria withEntityType()
    {
        return with(new EntityTypeSearchCriteria());
    }

    public PropertyTypeSearchCriteria withPropertyType()
    {
        return with(new PropertyTypeSearchCriteria());
    }

    public PropertyAssignmentSearchCriteria withPropertyAssignment()
    {
        return with(new PropertyAssignmentSearchCriteria());
    }

    public SemanticAnnotationSearchCriteria withOrOperator()
    {
        return (SemanticAnnotationSearchCriteria) withOperator(SearchOperator.OR);
    }

    public SemanticAnnotationSearchCriteria withAndOperator()
    {
        return (SemanticAnnotationSearchCriteria) withOperator(SearchOperator.AND);
    }

    @Override
    protected SearchCriteriaToStringBuilder createBuilder()
    {
        SearchCriteriaToStringBuilder builder = super.createBuilder();
        builder.setName("SEMANTIC_ANNOTATION");
        return builder;
    }

}

File: openbis/source/java/ch/systemsx/cisd/openbis/public/resources/api/v3/as/dto/semanticannotation/search/SemanticAnnotationSearchCriteria.js
define([ "require", "stjs", "as/dto/common/search/AbstractObjectSearchCriteria", "as/dto/common/search/PermIdSearchCriteria", "as/dto/entitytype/search/EntityTypeSearchCriteria",
		"as/dto/property/search/PropertyTypeSearchCriteria", "as/dto/property/search/PropertyAssignmentSearchCriteria", "as/dto/semanticannotation/search/PredicateOntologyIdSearchCriteria",
		"as/dto/semanticannotation/search/PredicateOntologyVersionSearchCriteria", "as/dto/semanticannotation/search/PredicateAccessionIdSearchCriteria",
		"as/dto/semanticannotation/search/DescriptorOntologyIdSearchCriteria", "as/dto/semanticannotation/search/DescriptorOntologyVersionSearchCriteria",
		"as/dto/semanticannotation/search/DescriptorAccessionIdSearchCriteria" ], function(require, stjs, AbstractObjectSearchCriteria) {
	var SemanticAnnotationSearchCriteria = function() {
		AbstractObjectSearchCriteria.call(this);
	};
	stjs.extend(SemanticAnnotationSearchCriteria, AbstractObjectSearchCriteria, [ AbstractObjectSearchCriteria ], function(constructor, prototype) {
		prototype['@type'] = 'as.dto.semanticannotation.search.SemanticAnnotationSearchCriteria';
		constructor.serialVersionUID = 1;
		prototype.withPermId = function() {
			var PermIdSearchCriteria = require("as/dto/common/search/PermIdSearchCriteria");
			return this.addCriteria(new PermIdSearchCriteria());
		};
		prototype.withEntityType = function() {
			var EntityTypeSearchCriteria = require("as/dto/entitytype/search/EntityTypeSearchCriteria");
			return this.addCriteria(new EntityTypeSearchCriteria());
		};
		prototype.withPropertyType = function() {
			var PropertyTypeSearchCriteria = require("as/dto/property/search/PropertyTypeSearchCriteria");
			return this.addCriteria(new PropertyTypeSearchCriteria());
		};
		prototype.withPropertyAssignment = function() {
			var PropertyAssignmentSearchCriteria = require("as/dto/property/search/PropertyAssignmentSearchCriteria");
			return this.addCriteria(new PropertyAssignmentSearchCriteria());
		};
		prototype.withPredicateOntologyId = function() {
			var PredicateOntologyIdSearchCriteria = require("as/dto/semanticannotation/search/PredicateOntologyIdSearchCriteria");
			return this.addCriteria(new PredicateOntologyIdSearchCriteria());
		};
		prototype.withPredicateOntologyVersion = function() {
			var PredicateOntologyVersionSearchCriteria = require("as/dto/semanticannotation/search/PredicateOntologyVersionSearchCriteria");
			return this.addCriteria(new PredicateOntologyVersionSearchCriteria());
		};
		prototype.withPredicateAccessionId = function() {
			var PredicateAccessionIdSearchCriteria = require("as/dto/semanticannotation/search/PredicateAccessionIdSearchCriteria");
			return this.addCriteria(new PredicateAccessionIdSearchCriteria());
		};
		prototype.withDescriptorOntologyId = function() {
			var DescriptorOntologyIdSearchCriteria = require("as/dto/semanticannotation/search/DescriptorOntologyIdSearchCriteria");
			return this.addCriteria(new DescriptorOntologyIdSearchCriteria());
		};
		prototype.withDescriptorOntologyVersion = function() {
			var DescriptorOntologyVersionSearchCriteria = require("as/dto/semanticannotation/search/DescriptorOntologyVersionSearchCriteria");
			return this.addCriteria(new DescriptorOntologyVersionSearchCriteria());
		};
		prototype.withDescriptorAccessionId = function() {
			var DescriptorAccessionIdSearchCriteria = require("as/dto/semanticannotation/search/DescriptorAccessionIdSearchCriteria");
			return this.addCriteria(new DescriptorAccessionIdSearchCriteria());
		};
	}, {});

	return SemanticAnnotationSearchCriteria;
})